File: android/app/src/main/java/com/brokenlunch/gr/data/model/Report.kt
package com.brokenlunch.gr.data.model

import com.squareup.moshi.Json
import com.squareup.moshi.JsonClass

@JsonClass(generateAdapter = true)
data class ReportRequest(
    @Json(name = "menu_item_id") val menuItemId: String,
    @Json(name = "reason") val reason: ReportReason,
    @Json(name = "comment") val comment: String? = null,
)

@JsonClass(generateAdapter = true)
data class ReportResponse(
    @Json(name = "id") val id: String,
    @Json(name = "status") val status: String,
    @Json(name = "menu_item_auto_disputed") val menuItemAutoDisputed: Boolean,
)

File: backend/app/schemas/report.py
"""Request/response models for POST /reports."""

from __future__ import annotations

from typing import Literal
from uuid import UUID

from pydantic import BaseModel, ConfigDict, Field

ReportReason = Literal[
    "wrong_price", "not_on_menu", "spam", "inappropriate", "other"
]


class ReportCreate(BaseModel):
    model_config = ConfigDict(extra="ignore")

    menu_item_id: UUID
    reason: ReportReason
    comment: str | None = Field(default=None, max_length=1000)


class ReportResponse(BaseModel):
    id: UUID
    status: str
    menu_item_auto_disputed: bool
